src/main/java/com/example/bookstoreapplication/controller/BookController.java

```diff
@@ -49,4 +51,11 @@ public BookDto updateBook(@PathVariable Long id, @RequestBody CreateBookRequestD
     public void delete(@PathVariable Long id) {
         bookService.delete(id);
     }
+
+    @GetMapping("/search")
+    public List<BookDto> search(@RequestParam(required = false) String[] titles,
+                                @RequestParam(required = false) String[] authors) {
+        BookSearchParameters searchParameters = new BookSearchParameters(titles, authors);

```

Comment:
you should be able to put that dto as a parameter, and request params must be mapped to that dto fields like that
```suggestion
    public List<BookDto> search(BookSearchParameters params) {
```